tests/unittests/InterpreterTest.cpp

```diff
@@ -896,3 +896,94 @@ TEST(Interpreter, nonLinearClassifier) {
     EXPECT_NEAR(RH.at({0, std::get<2>(tests[i])}), 1.0, 0.2);
   }
 }
+
+/// Generate images in two classes. A "line" is labeled as 0
+/// and a "cross" is labeled as 1
+void generateImageData(Tensor &images, Tensor &labels, unsigned numSamples, unsigned offset) {
+  auto L = labels.getHandle<size_t>();
+  auto Im = images.getHandle<>();
+  images.zero();
+
+  for (size_t i = 0; i < numSamples; i++) {
+    L.at({i, 0}) = i % 2;
+    size_t target = 2 + i % 10 + offset * 10;
+    for (size_t x = 0; x < 32; x++) {
+      for (size_t y = 0; y < 32; y++) {
+        if (i % 2 == 0) {
+          /// line
+          if (x == target)
+            Im.at({i, x, y, 0u}) = 1;
+        } else {
+          /// cross
+          if (x == target || y == target)
+            Im.at({i, x, y, 0u}) = 1;
+        }
+      }
+    }
+  }
+}
+
+TEST(Interpreter, simpleConv) {
+  unsigned numSamples = 50;
+  ExecutionEngine EE;
+
+  EE.getConfig().learningRate = 0.001;
+  EE.getConfig().momentum = 0.9;
+
+  auto &mod = EE.getModule();
+  Function *F = mod.createFunction("main");
+  F->setName("simpleConv");
+
+  auto *input = mod.createVariable(ElemKind::FloatTy, {numSamples, 32, 32, 1}, "input",
+                                   VisibilityKind::Public);
+
+  auto *ex = mod.createVariable(ElemKind::IndexTy, {numSamples, 1}, "exp",
+                                VisibilityKind::Public, Variable::TrainKind::None);
+  auto *CV0 = F->createConv("conv1", input, 16, 5, 1, 2, 1);
+  auto *RL0 = F->createRELU("relu1", CV0);
+  auto *MP0 = F->createPoolMax("pool1", RL0, 2, 2, 0);
+
+  auto *FCL1 = F->createFullyConnected("fc", MP0, 10);
+  auto *FCL2 = F->createFullyConnected("fc", FCL1, 2);
```

Comment:
You should probably either get rid of one of these FCs, or insert a nonlinearity (ReLU, maybe?) between them.  Since FCs are just matrix multiply, two in a row can be done with a single one.

Reply:
did that before, but same testing result.